app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,27 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(id: map_params[:id],loc: map_params[:loc])
+
+        if !@map.url.nil?
+          if @map.url.include? "http"
+            redirect_to @map.url
+          else
+            render plain: "#{@map.url}", status: 200
```

Comment:
The reserve locations are now tested in the controller spec, not the model, due to the possibility of the url variable containing a message and some simple logic around that in the controller.  So changing that would probably allow that test to go back into the model spec.  I agree that the message should probably be more than a plain text string, but I was going for Minimum Viable Product here.  So, for this one scenario, do you think we need to create a Map view in the Locations app?